### ISSUE TITLE
CBC 2.6.0 crashing

### ISSUE THREAD

**Issue by svigerske** [coin-or/Cbc#115]
Issue created by migration from Trac.

Original creator: dindy

Original creation time: 2011-07-08 12:59:53

Assignee: @tkralphs

Version: 2.6.0

Hello,

I have a model with wich CBC 2.6.0 crashes, the system doesn't let me attach it to the ticket. I'm using the precompiled CBC version available from CoinAll. I'm not able to compile a newer version so I don't know if this has been already fixed in the latest version.

The command line used is:

cbc -import crash.lp -solve -solu r.txt -quit

The CBC log up to the crash is:

----------
Cbc version 2.6.0, build Nov  7 2010
command line - cbc -import crash.lp -solve -solu r.txt -quit (default strategy 1)
Continuous objective value is -0.0845723 - 0.08 seconds
Cgl0003I 0 fixed, 0 tightened bounds, 14 strengthened rows, 0 substitutions
Cgl0004I processed model has 812 rows, 1088 columns (784 integer) and 80015 elements
Cbc0038I Pass   1: suminf.    0.10713 (2) obj. -0.0813099 iterations 218
Cbc0038I Solution found of -0.0810757
Cbc0038I Branch and bound needed to clear up 2 general integers
----------

Just contact me back if you need the model or some further help.

Bye,

**Comment by svigerske** [comment]
Comment by dindy created at 2011-07-18 20:17:45

Some further information. I have been able to compile CBC 2.7.0 using MinGW, compilation was smooth, no problems, no warnings. The problem persists. It works if I set presolve to off, but then I found another model where it crash even with presolve off. It works instead with preprocess set to off. The bug happens whenever the line:

Cbc0038I Branch and bound needed to clear up 2 general integers

appears in the log.

Hope it helps.

Bye

**Comment by svigerske** [comment]
Comment by dindy created at 2012-01-16 15:33:56

Retested recently with 2.7.5, same result. I have been able to narrow the problem down to a feasibility pump problem. By setting "-feasibilitypump off" it always works, though slower.

**Comment by svigerske** [comment]
Attachment [cbccrash.zip](https://www.coin-or.org/issue-attachments/Cbc/ticket115/cbccrash.zip) by dindy created at 2012-01-16 15:36:36

Sample lp model causing CBC to crash

**Comment by svigerske** [comment]
@jjhforrest With the current Cbc/master and the attached instance, I get a failing assertion here:
```
Welcome to the CBC MILP Solver 
Version: Trunk (unstable) 
Build Date: Mar 12 2019 
Revision Number: 2526 

command line - ./bin/cbc cbccrash.lp (default strategy 1)
Continuous objective value is -0.0845723 - 0.26 seconds
Cgl0003I 0 fixed, 0 tightened bounds, 14 strengthened rows, 0 substitutions
Cgl0004I processed model has 811 rows, 1087 columns (784 integer (571 of which binary)) and 80011 elements
Cbc0038I Initial state - 5 integers unsatisfied sum - 1.34248
Cbc0038I Pass   1: suminf.    0.10713 (2) obj. -0.0813099 iterations 17
Cbc0038I Solution found of -0.0810757
Cbc0038I Branch and bound needed to clear up 1 general integers
Cbc0038I Full problem 811 rows 1087 columns, reduced to 260 rows 262 columns
Cbc0038I Cleaned solution of -0.0795482
Cbc0038I Before mini branch and bound, 664 integers at bound fixed and 409 continuous of which 1 were internal integer and 0 internal continuous
Cbc0038I Full problem 811 rows 1087 columns, reduced to 8 rows 10 columns
Cbc0038I Mini branch and bound improved solution from -0.0795482 to -0.0824225 (1.46 seconds)
Cbc0038I Round again with cutoff of -0.0824891
Cbc0038I Reduced cost fixing fixed 167 variables on major pass 2
Cbc0038I Pass   2: suminf.    0.44417 (2) obj. -0.0825298 iterations 57
Cbc0038I Pass   3: suminf.    0.91765 (3) obj. -0.0824891 iterations 63
Cbc0038I Pass   4: suminf.    0.79104 (3) obj. -0.0824891 iterations 2
Cbc0038I Pass   5: suminf.    0.78449 (2) obj. -0.0824891 iterations 4
Cbc0038I Pass   6: suminf.    0.54559 (3) obj. -0.0824891 iterations 1
Cbc0038I Pass   7: suminf.    1.23300 (5) obj. -0.0824891 iterations 55
Cbc0038I Pass   8: suminf.    1.00345 (4) obj. -0.0824891 iterations 12
Cbc0038I Pass   9: suminf.    1.23300 (4) obj. -0.0824891 iterations 3
Cbc0038I Pass  10: suminf.    1.45725 (6) obj. -0.0824891 iterations 39
Cbc0038I Pass  11: suminf.    1.00345 (4) obj. -0.0824891 iterations 14
Cbc0038I Pass  12: suminf.    1.23300 (4) obj. -0.0824891 iterations 4
Cbc0038I Pass  13: suminf.    0.94318 (3) obj. -0.0824891 iterations 42
Cbc0038I Pass  14: suminf.    1.68049 (5) obj. -0.0824891 iterations 13
Cbc0038I Pass  15: suminf.    0.98945 (3) obj. -0.0824891 iterations 3
Cbc0038I Pass  16: suminf.    2.14719 (5) obj. -0.0824891 iterations 33
Cbc0038I Pass  17: suminf.    0.74095 (3) obj. -0.0824891 iterations 17
Cbc0038I Pass  18: suminf.    0.15161 (1) obj. -0.0824891 iterations 3
Cbc0038I Pass  19: suminf.    0.57281 (4) obj. -0.0824891 iterations 1
Cbc0038I Pass  20: suminf.    2.50313 (8) obj. -0.0824891 iterations 35
Cbc0038I Pass  21: suminf.    3.00458 (8) obj. -0.0824891 iterations 8
Cbc0038I Pass  22: suminf.    1.63599 (6) obj. -0.0824891 iterations 9
Cbc0038I Pass  23: suminf.    1.24115 (4) obj. -0.0824891 iterations 2
Cbc0038I Pass  24: suminf.    1.06995 (4) obj. -0.0824891 iterations 1
Cbc0038I Pass  25: suminf.    1.24115 (4) obj. -0.0824891 iterations 3
Cbc0038I Pass  26: suminf.    2.00345 (6) obj. -0.0824891 iterations 39
Cbc0038I Pass  27: suminf.    2.00345 (6) obj. -0.0824891 iterations 12
Cbc0038I Pass  28: suminf.    1.33679 (5) obj. -0.0824891 iterations 5
Cbc0038I Pass  29: suminf.    1.23300 (4) obj. -0.0824891 iterations 1
Cbc0038I Pass  30: suminf.    1.00345 (4) obj. -0.0824891 iterations 1
Cbc0038I Pass  31: suminf.    1.23300 (4) obj. -0.0824891 iterations 3
Cbc0038I Rounding solution of -0.0824526 is better than previous of -0.0824225

Cbc0038I Before mini branch and bound, 569 integers at bound fixed and 404 continuous
Cbc0038I Full problem 811 rows 1087 columns, reduced to 23 rows 22 columns
Cbc0038I Mini branch and bound improved solution from -0.0824526 to -0.0826911 (2.39 seconds)
Cbc0038I Round again with cutoff of -0.0827605
Cbc0038I Reduced cost fixing fixed 184 variables on major pass 3
Cbc0038I Pass  31: suminf.    0.62587 (4) obj. -0.0827605 iterations 60
Cbc0038I Pass  32: suminf.    1.13082 (4) obj. -0.0827605 iterations 30
Cbc0038I Pass  33: suminf.    1.13082 (4) obj. -0.0827605 iterations 0
Cbc0038I Pass  34: suminf.    0.65712 (2) obj. -0.0827605 iterations 2
Cbc0038I Pass  35: suminf.    0.56805 (4) obj. -0.0827605 iterations 1
Cbc0038I Pass  36: suminf.    0.65712 (3) obj. -0.0827605 iterations 29
cbc: ../../../Cbc/src/CbcHeuristic.cpp:2597: virtual int CbcRounding::solution(double&, double*, double): Assertion `fabs(floor(value + 0.5) - value) <= integerTolerance' failed.
```
Can you have a look?

When building without assertions, then Cbc/master seems to run fine (no crash).

**Comment by jjhforrest** [comment]
Modified code slightly - should be Ok

**Comment by svigerske** [comment]
Also this fix seems to have been lost. Can you find it again?
Currently, the mirroring of svn to github is still on (though it sometimes seems to get stucked), so changes to GitHub are overwritten (more or less) regularly.

**Comment by svigerske** [comment]
Fixed by 75e9fdb2b6788.
